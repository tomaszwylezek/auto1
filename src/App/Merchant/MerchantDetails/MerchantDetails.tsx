import React, { FC } from 'react';

// Features
import { Button } from 'shared/Button/Button';
import { Bid } from 'App/Merchant/MerchantDetails/Bid/Bid';

// Models
import { IMerchantDetailsProps } from 'App/Merchant/MerchantDetails/MerchantDetails.types';
import { ButtonTheme, IBid, Size } from 'model';

export const MerchantDetails: FC<IMerchantDetailsProps> = ({ id, phone, name, bids, onDeleteClick }): JSX.Element => (
  <div className="card my-2" key={id}>
    <div className="card-header py-1 px-3">
      <div className="d-flex justify-content-between">
        {name}

        <div>
          <Button theme={ButtonTheme.outlineSecondary} size={Size.sm} to={`/${id}`} className="mr-2">
            Edit
          </Button>
          <Button theme={ButtonTheme.danger} size={Size.sm} onClick={onDeleteClick}>
            Delete
          </Button>
        </div>
      </div>
    </div>
    <div className="card-body p-2">
      <div>Contact: {phone}</div>
      <div className="d-flex">
        {bids
          .sort((a: IBid, b: IBid): number => +new Date(b.created) - +new Date(a.created))
          .map(({ carTitle, id, amount, created }: IBid) => (
            <Bid key={id} id={id} amount={amount} carTitle={carTitle} created={created} />
          ))}
      </div>
    </div>
  </div>
);
