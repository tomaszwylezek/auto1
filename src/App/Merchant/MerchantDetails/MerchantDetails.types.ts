import { IBid } from 'model';

export interface IMerchantDetailsOwnProps {
  id: string;
  name: string;
  phone?: string;
  onDeleteClick: () => void;
  bids: IBid[];
}

export interface IMerchantDetailsConnectedState {}

export interface IMerchantDetailsConnectedActions {}

export interface IMerchantDetailsProps
  extends IMerchantDetailsConnectedState,
    IMerchantDetailsConnectedActions,
    IMerchantDetailsOwnProps {}

export interface IMerchantDetailsState {}
