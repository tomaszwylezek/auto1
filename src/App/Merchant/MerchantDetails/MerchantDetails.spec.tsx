import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { MerchantDetails } from './MerchantDetails';

// Models
import { IMerchantDetailsProps, IMerchantDetailsState } from './MerchantDetails.types';

describe('MerchantDetails component should', (): void => {
  let props: IMerchantDetailsProps;
  let component: ShallowWrapper<IMerchantDetailsProps, IMerchantDetailsState>;

  beforeEach((): void => {
    props = {};

    component = shallow(<MerchantDetails {...props} />);
  });

  it('render correctly', (): void => {
    expect(component).toMatchSnapshot();
  });
});
