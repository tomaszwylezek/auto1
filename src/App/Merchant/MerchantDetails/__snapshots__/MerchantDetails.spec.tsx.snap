// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MerchantDetails component should render correctly 1`] = `
<div
  className="card my-2"
  key="123"
>
  <div
    className="card-header py-1 px-3"
  >
    <div
      className="d-flex justify-content-between"
    >
      Foo Bar
      <div>
        <Button
          className="mr-2"
          size="sm"
          theme="outline-secondary"
          to="/123"
        >
          Edit
        </Button>
        <Button
          onClick={[MockFunction]}
          size="sm"
          theme="danger"
        >
          Delete
        </Button>
      </div>
    </div>
  </div>
  <div
    className="card-body p-2"
  >
    <div>
      Contact: 
      123
    </div>
    <div
      className="d-flex"
    >
      <Bid
        amount={12}
        carTitle="Toyota"
        created="Sat Jul 06 2019 10:31:54 GMT+0200 (Central European Summer Time)"
        id="3"
        key="3"
      />
    </div>
  </div>
</div>
`;
